localstack-core/localstack/services/lambda_/invocation/event_manager.py

```diff
@@ -57,6 +58,11 @@ class SQSInvocation:
     exception_retries: int = 0
 
     def encode(self) -> str:
+        # Encode TraceHeader
+        aws_trace_header_str = TraceHeader(
```

Comment:
@dfangl not super happy about this encoding/decoding hack. Feel free to share your suggestion.

This doesn't go well if `aws_trace_header` is not of type `TraceHeader`